src/mon/OSDMonitor.cc

```diff
@@ -662,6 +645,41 @@ void OSDMonitor::create_pending()
 	      << pending_inc.new_nearfull_ratio << dendl;
     }
   }
+
+  // Rewrite CRUSH rule IDs if they are using legacy "ruleset"
+  // structure.  Only do this if there is no pending crush map or pool
+  // creations/deletions/updates and we don't have to worry about
```

Comment:
There doesn't seem to be any logic enforcing this constraint that we not have other stuff going on?

Reply:
oops, that comment is obsolete from when this code was elsewehre.  it's not in create_pending(), so by definition there is nothing (yet) going on.  fixed